metis-processor/src/main/java/eu/europeana/metis/processor/utilities/ImageEnhancerUtil.java

```diff
@@ -13,227 +16,338 @@
-                        final long startTime = System.nanoTime();
-                        byte[] enhancedImage = enhanceImage(largeThumbnail, largeThumbnailObjectName);
-//                        Files.write(Paths.get("/tmp/test_thumbnails/enhanced-" + largeThumbnailObjectName), enhancedImage);
-                        final long elapsedTimeInNanoSec = System.nanoTime() - startTime;
-
-                        List<Thumbnail> thumbnails = generateThumbnails(webResourceWrapper, largeThumbnailObjectName, enhancedImage);
-                        uploadThumbnails(webResourceWrapper, thumbnails);
-                        appendSuccessReport(webResourceWrapper, enhancedImage, thumbnailHex, elapsedTimeInNanoSec, reportRow);
-                    } else {
-                        appendProcessedReport(webResourceWrapper, thumbnailHex, reportRow);
-                    }
-                }
-            } catch (MediaExtractionException | IOException e) {
-                LOGGER.error("Failed enhancing thumbnail image {} {}", webResourceWrapper.getAbout(), e);
-                appendFailReport(reportRow);
+  private void extractThumbnailEnhanceAndReUploadIt(RDF recordToProcess) {
+    List<Aggregation> curatedAggregationList = Optional.ofNullable(recordToProcess.getAggregationList())
+                                                       .orElse(Collections.emptyList());
+    final String recordId = recordToProcess.getProvidedCHOList().get(0).getAbout();
+    if (hasAggregationWithHasView(curatedAggregationList)) {
+      LOGGER.info("=>hasView");
+      curatedAggregationList.forEach(aggregationItem -> {
+            if (hasShownBy(aggregationItem)) {
+              LOGGER.info("=>shownByItem");
+              processShownBy(recordToProcess, aggregationItem, recordId);
             }
-        });
+            LOGGER.info("=>hasViewItem");
+            processHasView(recordToProcess, aggregationItem, recordId);
+          }
+      );
+    } else {
+      processSimpleEntry(recordToProcess, recordId);
     }
+  }
 
-    private boolean isWebResourceResolutionAndImageResolutionSame(WebResourceWrapper webResourceWrapper, byte[] imageBytes) throws IOException {
-        final BufferedImage largeThumbnailBufferedImage = getBufferedImage(imageBytes);
-        if (largeThumbnailBufferedImage == null) {
-            throw new IOException("Could not get resolution from provided image");
-        }
-        return largeThumbnailBufferedImage.getHeight() == webResourceWrapper.getHeight() &&
-                largeThumbnailBufferedImage.getWidth() == webResourceWrapper.getWidth();
-    }
+  private void processSimpleEntry(RDF recordToProcess, String recordId) {
+    List<WebResourceWrapper> webResourceWrappers = new RdfWrapper(recordToProcess)
+        .getWebResourceWrappers(Set.of(WebResourceLinkType.IS_SHOWN_BY))
+        .stream()
+        .filter(webResourceWrapper -> webResourceWrapper != null && webResourceWrapper.getMimeType() != null)
```

Comment:
If you check the WebResourceWrapper.getMimeType it already checks internally for nullity of mimetype

Reply:
mm, strange, I got `nullpointerexception` let me recheck this code again.

Reply:
in the end it `WebResourceWrapper.getMimeType` has `.orElse((Object)null)` so when consuming `getMimeType().method()` this causes an null pointer exception. I will leave the validation.

Reply:
Yes you are right!